newa/cli.py

```diff
@@ -194,50 +193,99 @@ def cmd_event(ctx: CLIContext, errata_ids: list[str]) -> None:
 def cmd_jira(ctx: CLIContext, issue_config: str) -> None:
     ctx.enter_command('jira')
 
-    # this is here temporarily so we generate fake Jira issue IDs
-    jira_id_gen = itertools.count(start=1)
+    # Get control of Jira.
+    # TODO: Hardcoded project for now.
+    # It will be served from the component-config.yaml later.
+    jira = JiraHandler("BASEQESEC")
 
     for erratum_job in ctx.load_erratum_jobs('event-'):
         # read Jira issue configuration
         config = ErratumConfig.from_yaml_file(Path(os.path.expandvars(issue_config)))
 
-        # TODO: record created/existing issues. Instead of `Any`, maybe something
-        # from the Jira library would be stored. Or just a Jira ticket ID.
-        known_issues: dict[str, Any] = {}
+        # Record created/existing issues.
+        known_issues: dict[str, Issue] = {}
 
         # Iterate over issue actions. Take one, if it's not possible to finish it,
         # put it back at the end of the queue.
         issue_actions = config.issues[:]
 
         while issue_actions:
             action = issue_actions.pop(0)
+            rendered_summary = render_template(action.summary, ERRATUM=erratum_job)
+            rendered_description = render_template(action.description, ERRATUM=erratum_job)
+            rendered_assignee = render_template(action.assignee, ERRATUM=erratum_job)
 
+            # For debugging.
             print(f'* Would create a {action.type.name} issue:')
             print(f'     summary: {action.summary}')
             print(f'     summary: {action.description}')
             print()
 
-            if action.id in known_issues:
-                raise Exception(f'Issue "{action.id}" is already created!')
-
+            # Find existing issues in Jira related to erratum_job and action.id.
+            #
+            # We re looking for Jira issues such that:
+            #
+            # 1) summary matches rendered_summary (e.g. Errata respin 0) and
+            # 2) description contains erratum_job.erratum (e.g. Erratum(130704) and
+            # 3) description contains erratum_job.erratum.release (e.g. Release(RHEL-8.8.0.Z.EUS))
+            query = f""" summary ~ '{rendered_summary}' and
+                         description ~ '{erratum_job.erratum!r}' and
+                         description ~ '{erratum_job.erratum.release!r}' """
+            for jira_issue in jira.search_open_issue(query):
+                desc = jira_issue.fields.description
+
+                # We found relevant issue in Jira, it is either related to the
+                # current respin or not.
+                #
+                # 1) If it is, it is not created again - it is skipped.
```

Comment:
I think we will need to modify issue processing a bit so that at first we spit issues into  categories "current" and "old" and then process each category individually. The reason for that is that when closing "old" issue we would most likely want to add there a comment referencing the "current" one.

Reply:
I'll take a look into this.

Reply:
Sounds reasonable but the split might complicate code a bit. I'll keep the original version commented out above the new one so that you can still keep eye on the original one. The thing is that I still need to go through the issue actions and 

- I need to remember ones for which I need to create issues  (in the subsequent loop)
- I need to find old issues and associate them with their issue action so that I know what to do with then (in the subsequent loop

So I will end up with three loops:

1. The original one making the split.
2. The one for creating issues.
3. The one for processing issue actions for old issues.

Reply:
I think se vám process issue config one by one. If it refers a parent se dont know, push it at the end of the queue, if not, we can proceed  handling the Jira content. Just realized, when pushing item at the end, we need to add some sort of Infinite loop detection.

Reply:
Please check the current version of the command. Here's what I did:

1) _First, go through issue actions to see what issues from the previous respin(s) need to be processed._

The sole purpose of this loop is to find outdated issues and store them with their associated on_respin action.

2) _Second, iterate over issue actions, take one, if it's not possible to finish it, put it back at the end of the queue. Create and store issue otherwise._

This is basically the original loop - we create issues if we can or put action to the end and get back to it again later. endless loop is detected too.

3) _Third, iterate over issue actions for issues related to the previous respin(s) and act accordingly._

Handle the outdated issues detected in the first step. based on their on_respin action.